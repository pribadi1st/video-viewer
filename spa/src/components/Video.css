.video {
  display: inline-block;
  margin: 0.5rem;
}

.video-player {
  max-width: 15rem;
  max-height: 15rem;
}
