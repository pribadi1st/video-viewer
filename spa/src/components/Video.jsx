import React from "react";

import "./Video.css";

const Video = (props) => {
  const { url, title, thumbnail } = props;

  return (
    <div className="card video">
      <video
        controls
        src={url}
        preload="none"
        poster={thumbnail}
        title={title}
        className="video-player"
      />

      <div className="card-body">
        {title}
      </div>
    </div>
  );
};

export default Video;
