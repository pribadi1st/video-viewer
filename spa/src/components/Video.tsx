import React from "react";

import "./Video.css";
interface VideoProps {
  url: string
  title: string
  thumbnails: {
    small: string
    medium: string
    large: string
  }
  category: {
    text: string
    value: number
  }
}

const Video = (props:VideoProps) => {
  const { url, title, thumbnails, category } = props;
  const [show, setShow] = React.useState(false);
  const showTitle = (state: boolean = false) => {
    setShow(state)
  }

  return (
    <div
      className="card video"
      onMouseEnter={() => showTitle(true)}
      onMouseLeave={() => showTitle(false)}
    >
      <video
        controls
        src={url}
        preload="none"
        title={title}
        className="video-player"
        poster={thumbnails.large}
      />
      {
        show && 
          <div className="position-absolute top-0 start-0 bg-light w-100">
            <p className="mt-3">
              {title}
            </p>
          </div>
      }

      <div className="card-body">
        {category.text}
      </div>
    </div>
  );
};

export default Video;
