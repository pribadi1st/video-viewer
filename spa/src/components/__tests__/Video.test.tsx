import React  from "react"
import { render, screen, cleanup } from "@testing-library/react"
import renderer from 'react-test-renderer'
import Video from "../Video"



// Fix the test in this component, The test are failing right now.

afterEach(()=> {
    cleanup();
})

const videoTestProps = { url:  'https://www.caspar-health.com', title: "shoulder exercise" }


test('should render video component', ()=> {
    render(<Video video= {videoTestProps}/>)  
    const videoElement = screen.getByTestId("video-card");
    expect(videoElement).toBeInTheDocument();
    expect(videoElement).toHaveLength(3)
}) 


test('matches the snapshot', ()=> {
    const tree = renderer.create(<Video video={videoTestProps} />)
    expect(tree).toMatchSnapshot();
})