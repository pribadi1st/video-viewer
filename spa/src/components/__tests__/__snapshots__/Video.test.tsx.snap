// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="card video"
>
  <video
    className="video-player"
    controls={true}
    preload="none"
  />
  <div
    className="card-body"
  />
</div>
`;
